app/models/project.rb

```diff
@@ -10,13 +10,9 @@ def maya_top_level_dir
     end
   end
 
-  def initialize
-    super
-    write_attribute(:directory, Project.maya_top_level_dir)
-  end
-
-  def initialize(params)
+  def initialize(params = {})
     super
+    write_attribute(:directory, params[:directory] ||= Project.maya_top_level_dir)
```

Comment:
what is the purpose of this?

Reply:
This initializes the directory to the ~/projects/maya.  I figured from the other comment, you'd rather the initial values be set in the model instead of the controller. 

The initializer api has to take in params because that's how rails calls it in different places.

Reply:
Setting initial values in the model is ideal. For setting initial values in Rails, I think maybe default values in getters would be preferable, with utilizing the Rails callback to set initial values. That way you could avoid overwriting the initializer. In this case, an example getter would be:

```ruby
def directory
  attributes[:directory] || Project.maya_top_level_dir
end
```

But in this case, the whole idea of setting a value in the initializer is so that the view will act in a certain way (the file picker in this case). So this is technically a hack, because it doesn't make any sense that the default project directory for an instance of project would be `~/maya/projects` which is definitely not a Maya project directory. If its necessary to introduce hacks, please add a big comment prefixed with `# HACK:` and explain the hack.

Based on Slack chats Friday there is likely a solution that can avoid the need for this hack.